src/nrncvode/netcvode.cpp

```diff
@@ -5560,7 +5560,7 @@ static void* eval_cond(NrnThread* nt) {
 }
 void Cvode::evaluate_conditions(NrnThread* nt) {
     if (!nt) {
-        if (nrn_nthread > 1) {
+        if (nrn_nthread > 1 && nctd_ > 1) {
```

Comment:
cc: @nrnhines, I'm sure this was a fix for a TSan error but I don't immediately have the error to hand. I can try and reproduce it if needed.

Reply:
I can see why tsan would complain about this for the case (I presume never occurring) of  NULL nt, multiple threads, and the possibility that nctd_ == 1. The whole point ( I hypothesize:)) of the fragment is to (one level recursive) turn the call into separate calls to the same cvode instance but use distinct CvodeThreadData for each thread. Otherwise there would be a race in regard to using the single CvodeThreadData[0] given
```
#define CTD(i) ctd_[((nctd_ > 1) ? (i) : 0)]
```
But if it ever happens that the arg nt is NULL, nrn_nthread > 1, and nctd == 1, then I'd like to analyze it as it would seem to be a real bug. (and I wonder if we should instead assert that ```nctd_ == nrn_nthread``` before calling the nrn_multithread_job) Otherwise it would seem to be possible to evaluate_conditions for some threads but not all.

Reply:
> But if it ever happens that the arg nt is NULL, nrn_nthread > 1, and nctd == 1, then I'd like to analyze it as it would seem to be a real bug. 

That is the case for ```fast/t18.hoc``` for the local variable step method. Definitely a bug. ```if (nrn_nthread > 1 && nctd_ > 1) {``` fixed it.  In the local variable step case nctd is 1 and in the global variable case, nctd is nrn_nthread.